src/toxcore/dht/server.rs

```diff
@@ -276,6 +296,7 @@ impl Server {
                 };
                 let mut state = self.state.write().expect("Failed to obtain ServerState lock");
                 state.kbucket.try_add(&packed_node);
+                client.last_resp_time = Instant::now();
```

Comment:
How do you update the client? Since you get only `mut Client` not `&mut Client`, is is not updated in you `peer_cache`.